code/services/SAMLConfiguration.php

```diff
@@ -51,7 +51,7 @@ public function asArray()
             'url' => $sp['entityId'] . '/saml/acs',
             'binding' => OneLogin_Saml2_Constants::BINDING_HTTP_POST
         ];
-        $conf['sp']['NameIDFormat'] = OneLogin_Saml2_Constants::NAMEID_TRANSIENT;
+        $conf['sp']['NameIDFormat'] = isset($sp['nameIdFormat') ? $sp['nameIdFormat'] : OneLogin_Saml2_Constants::NAMEID_TRANSIENT;
```

Comment:
Indeed. Fixed now; I'd altered this file separately after experimenting with my development instance. Trued-up now.